app/frontend/components/domains/permit-application/revision-sidebar.tsx

```diff
@@ -79,23 +114,35 @@ export const RevisionSideBar = observer(
     const handleOpenRequestRevision = async (event, upToDateFields) => {
       if (!permitApplication.formJson) return
 
-      const foundRevisionRequest = upToDateFields.find((field) => field.requirementJson?.key === event.detail.key)
+      const finder = (rr) => rr.requirementJson?.key === event.detail.key
+      const foundRevisionRequestDefault =
+        isViewingPastRequests && selectedPastSubmissionVersion?.revisionRequests?.find(finder)
+      const foundRevisionRequest = upToDateFields.find(finder)
       const foundRequirement = getRequirementByKey(permitApplication.formJson, event.detail.key)
       const foundSubmissionJson = getSinglePreviousSubmissionJson(permitApplication.submissionData, event.detail.key)
 
       setRevisionRequest(foundRevisionRequest)
+      setRevisionRequestDefault(foundRevisionRequestDefault)
       setRequirementForRevision(foundRequirement)
       setSubmissionJsonForRevision(foundSubmissionJson)
       onOpen()
     }
 
     useEffect(() => {
       const handleOpenEvent = (event) => handleOpenRequestRevision(event, fields)
+      // Listener needs to be re-registered every time the tab index changes
+      document.removeEventListener("openRequestRevision", handleOpenEvent)
```

Comment:
you shouldn't need to do this, the cleanup function should take care of it. This line should essentially be doing nothing. The remove event listener uses a combination of the event name and the callback function to determine which handler to remove. But this `handleOpenEvent` has a new reference in the render, so  `document.removeEventListener("openRequestRevision", handleOpenEvent)` in the first line is not removing anything as an event listener with this reference was not set in this render cycle.

Please let me know if you would like to go over this comment on a call